### PR TITLE
Move dotnet server into vscode repo and connect language client

### PR DESCRIPTION
**Pull request by gracegoo-stripe**
This PR introduces the dotnet language server and connects the language client to it. This is not a complete implementation, below is a checklist of things left TODO. This will also merge into a dotnet-server branch until everything is done.

## APIReferenceHandler
This is the main logic that handles what content to return when the user hovers over a document we are interested in. We are using the semantic model to get the fully name-spaced type of the method invocation. Previously, I had tried to do this with the syntax tree which also worked but we had to handle all the different syntax for a method invocation. For example, we had to handle each of these cases ourselves: `priceService.GetAsync()`, `new PriceService().GetAsync()`, etc. The semantic model is the compiled version of the code so we can simply extract the details we need from the symbol once we find it. With this method, we are blindly trying to get the type and name of the symbol, even if it may not be a method invocation. The key is then used to lookup the api reference path in the configuration file and returns a null Hover instance if it cannot be found, and a real hover instance if it is found. 

Demo: 
https://user-images.githubusercontent.com/75757829/119735293-f569ee80-be30-11eb-9612-651c80889412.mov

## Checklist
- [x] Set up dotnet server and connect it to the extension
- [x] Implement API reference handler with semantic model analysis. 
- [ ] Implement Text document hander (I had previously done this with a buffer manager, but I want to see if I can add changes directly to the MSBuildWorkspace instance.
- [ ] Unit Testing
- [ ] Channel logs to the output channel of the extension -- right now only errors are getting through.
- [ ] Client-side: Add conditional server startup logic based on workspace project 
- [ ] Client-side: Update dotnet executable fetching -- right now it's hard-coded 
- [ ] Get the project file from the client
- [ ] Update debugging steps since we've merged the repos
- [ ] Update the configuration file with real entries
- [ ] Webpack building?

### CODE REVIEW

File: src/stripeLanguageServer/client.ts [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 
 export class StripeLanguageClient {
-  static activate(context: ExtensionContext, serverOptions: ServerOptions, telemetry: Telemetry) {
-    const clientOptions: LanguageClientOptions = {
+  static activate(
+    context: vscode.ExtensionContext,
+    serverOptions: ServerOptions,
+    telemetry: Telemetry,
+  ) {
+    const outputChannel = vscode.window.createOutputChannel('Stripe Lanaguage Client');
```
Comment (vcheung-stripe) [inline src/stripeLanguageServer/client.ts:21]:
```suggestion
    const outputChannel = vscode.window.createOutputChannel('Stripe Language Client');
```

File: src/stripeLanguageServer/client.ts [hunk 2]
```diff
@@ -18,22 +32,82 @@ export class StripeLanguageClient {
...
+          return ErrorAction.Continue;
+        },
+        closed: () => CloseAction.Restart,
+      },
+      revealOutputChannelOn: RevealOutputChannelOn.Error,
+    };
+
+    // Create the language client and start the client.
+    const dotnetClient = new LanguageClient(
+      'stripeCsharpLangaugeServer',
```
Comment (vcheung-stripe) [inline src/stripeLanguageServer/client.ts:96]:
```suggestion
      'stripeCsharpLanguageServer',
```

File: src/stripeDotnetLanguageServer/Program.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                                new ConfigurationItem
+                                {
+                                    Section = "typescript",
+                                }
+                            ).AddSingleton(
+                                new ConfigurationItem
+                                {
+                                    Section = "terminal",
+                                }
+                            );
```
Comment (vcheung-stripe) [inline src/stripeDotnetLanguageServer/Program.cs:84]: What are these ConfigurationItems used for?

**Review by vcheung-stripe — COMMENTED** [review]
Makes sense to me after your explanation. The SemanticModel looks perfect for our use case. Just a couple nits and a question.

File: src/stripeLanguageServer/client.ts (hunk 1 above)
Comment (gracegoo-stripe) [reply src/stripeLanguageServer/client.ts:21]: Oops, me and my typos .. Updating!

File: src/stripeDotnetLanguageServer/Program.cs (hunk 3 above)
Comment (gracegoo-stripe) [reply src/stripeDotnetLanguageServer/Program.cs:80]: Oops, leftovers from the [sample](https://github.com/OmniSharp/csharp-language-server-protocol). I think these are used to send over configs from the client -- let me remove these. Good catch!

**Comment by gracegoo-stripe** [comment]
Addressed comments, and I also added another commit that fixes our intellisense issue for the subdirectory.

**Comment by gracegoo-stripe** [comment]
Also a note for the build failures -- those will be fixed when I figure out how to get the dotnet executable for each platform

**Review by vcheung-stripe — APPROVED** [review]
Makes sense that the action fails. This LGTM.

**Pull request merged**